runtime/src/snapshot_utils.rs

```diff
@@ -39,54 +39,69 @@ use {
     thiserror::Error,
 };
 
+/// Trait to query the snapshot archive information
+pub trait SnapshotArchiveInfo {
+    fn snapshot_archive_info_inner(&self) -> &SnapshotArchiveInfoInner;
+
+    fn path(&self) -> &PathBuf {
+        &self.snapshot_archive_info_inner().path
+    }
+
+    fn slot(&self) -> Slot {
+        self.snapshot_archive_info_inner().slot
+    }
+
+    fn hash(&self) -> &Hash {
+        &self.snapshot_archive_info_inner().hash
+    }
+
+    fn archive_format(&self) -> ArchiveFormat {
+        self.snapshot_archive_info_inner().archive_format
+    }
+}
+
 /// Common information about a snapshot archive
-#[derive(PartialEq, Eq, Debug)]
-struct SnapshotArchiveInfo {
+#[derive(PartialEq, Eq, Debug, Clone)]
+pub struct SnapshotArchiveInfoInner {
```

Comment:
hmmm I think I like the `SnapshotArchiveInfo` name here, and should instead rename the trait

Reply:
OK, will do!

Reply:
Done!